syntax = "proto3";
package myapp;

message MyRequest {
  enum CommandType {
    CREATE_USER = 0;
    DELETE_USER = 1;
    UPDATE_USER = 2;
    LIST_USERS  = 3;
    AUTHENTICATE = 4;
  }
  CommandType cmd = 1;
  string username = 2;
  string password = 3;
  bool   is_admin = 4;
  string new_username = 5;
}

message MyResponse {
  bool success = 1;
  string error = 2;
  repeated string users = 3;
  bool is_admin = 4;
}
